### ISSUE TITLE
Possibly necessary consideration of model object creation and handling?

### ISSUE THREAD

**Issue by hideya** [kgoedecke/loopback-swift-crud-example#2]
It seems to me that the intended way to create a new model object is something like the following:

```
let newWidget = repository.modelWithDictionary(nil) as! Widget
```

instead of `let newWidget = Widget()` found at [Line 33 of WidgetViewController.swift](https://github.com/kgoedecke/loopback-swift-crud-example/blob/master/loopback-swift-crud-example/WidgetViewController.swift#L33) (although the above looks a bit too wordy and I think the API needs refinements).

If a model is created in the above way, we can call `newWidget.saveWithSuccess(...)` directly, instead of converting a model to a dictionary and then recreate a model ([MyTableViewController.swift#L99-L100](https://github.com/kgoedecke/loopback-swift-crud-example/blob/master/loopback-swift-crud-example/MyTableViewController.swift#L99-L100) and [LBRepositoryController.swift#L64](https://github.com/kgoedecke/loopback-swift-crud-example/blob/master/loopback-swift-crud-example/LBRepositoryController.swift#L64))

Also, if a model is returned from the framework, I think we can call `destroyWithSuccess()` directly without performing `findById()` as seen at [LBRepositoryController.swift#L103](https://github.com/kgoedecke/loopback-swift-crud-example/blob/master/loopback-swift-crud-example/LBRepositoryController.swift#L103)

I'm wondering if the above points are possibly something better be considered.  Could you share some thoughts on this?

**Comment by kgoedecke** [comment]
@hideya this looks very interesting, when I was implementing the example I was already wondering if there is a more convenient way to create and destroy models. I'll check your suggestions out and let you know what I think asap. Thanks for sharing this!

**Comment by kgoedecke** [comment]
@hideya So I looked into this and the following problem came up: 
When I try to cast from LBPersistedModel to Widget (for example when creating new Widget) then I lose the obj-c ._id attribute. This leads to errors when executing destroy or update. I'm wondering why I lose this attribute when casting. Do you have any idea why this might occur? I think this might be happening because swift doesn't use the _ anymore for attributes. I can see id being set for my Widget objects but not _id.

**Comment by hideya** [comment]
@kgoedecke This is weird...  I tried `let newWidget = repository.modelWithDictionary(nil) as! Widget` in your codebase.  It worked fine when to invoke `saveWithSuccess`, but an issue came up when to invoke `destroyWithSuccess` as you mentioned.  It crashed with the following assertion error:

```
*** Terminating app due to uncaught exception 'NSInternalInconsistencyException', reason: 'Invalid repository.'
```

This means that the `widget` object's `repository` parameter is `nil`.  However, I did make sure that `repository` is not `nil` when to append the widget to the array at [this line](https://github.com/kgoedecke/loopback-swift-crud-example/blob/master/loopback-swift-crud-example/MyTableViewController.swift#L103).  The `repository` property is set by the `repository.modelWithDictionary` call and shouldn't get unset later on.  However, it seems to get cleared later on somehow...  I did't have time to dig this symptom deeper...

> I can see id being set for my Widget objects but not _id.

This sounds strange...  When I tried your codebase, I was able to do like `print(model._id)`, whereas `model.id` causes a syntax error 'Value of type 'LBPersistedModel' has no member 'id''

As to the `._id` property of a newly created model, please note that it is initially `nil` and gets set as a result of invoking `saveWithSuccess`.  As far as I could tell, it gets set correctly in your codebase and I didn't find any issue regarding it so far.  Would you please double-check whether the errors when to execute destroy or update are caused by an unset `._id` or by the 'Invalid repository' situation mentioned above?

I've updated https://github.com/hideya/loopback-swift-simple-example to perform model creation and deletion in the manner proposed above (and updated its model management behaviour closer to your app), but I don't encounter any issue there...  Would you have any thoughts why it works OK on my side but causes the error on your side...?

**Comment by kgoedecke** [comment]
@hideya I updated my repository and now everything works just like in your example! I'll also update the tutorial on my blog. All issues with the `._id` did disappear as well. Again thanks a lot for your contribution.

**Comment by hideya** [comment]
@kgoedecke I'm glad to know that you've made it working!  Thank YOU for sharing the great example and such a helpful tutorial first of all.  I'm happy that I was able to contribute to it! :-)
